src/Service/Resolvers/SqlMutationEngine.cs

```diff
@@ -331,13 +331,22 @@ public class SqlMutationEngine : IMutationEngine
 
                 if (context.OperationType is Config.Operation.Insert)
                 {
-                    if (mutationResultRow is null || mutationResultRow.Columns.Count == 0)
+                    if (mutationResultRow is null)
                     {
                         // this case should not happen, we throw an exception
                         // which will be returned as an Unexpected Internal Server Error
                         throw new Exception();
                     }
 
+                    if (mutationResultRow.Columns.Count == 0)
+                    {
+                        throw new DataApiBuilderException(
+                            message: "Could not insert row with given values.",
+                            statusCode: HttpStatusCode.Forbidden,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed
```

Comment:
Is AuthorizationCheckFailed the only reason why insert row columns is 0 if we haven't errored out earlier?

Reply:
Yeah if there were no exceptions encountered in the insert operation on the database and still the result that we have does not contain any row, then no row was inserted. This happens only when we insert nothing - because of database policy.